UPGRADE.md

```diff
@@ -7,6 +7,12 @@
 The `Router` service now interprets more into the URL parameters that are passed to it. In addition to parsing numbers,
 boolean and strings to its JS equivalents, it will now do the same for `undefined` and dates in the format `yyyy-mm-dd`.
 
+### Passing MultiSelectionStore to MultiAutoComplete
```

Comment:
Well, if we leave it like it is, it is inconsistent view the `List`, `Form`, `SmartContent` and `MediaItem` components, which all receive a store, instead of creating them on their own. So the consistency argument can't really apply here IMO :see_no_evil: 

If we want to be consistent, I would prefer to always pass the store instead of creating it in the component. This would allow to have more control over it, and reuse it when applicable. The other options are:

1. Accept double requests for the same date e.g. in the Organization selection when filter in #5035.
2. Use the `MultiAutoComplete` component as you've suggested and duplicate the code of creating the `SearchStore`, updating the value when some props change, handling the text input, ...

I would clearly vote for option 1 (as implemented in here), because option 2 has serious drawbacks of duplicating code multiple times. I would also change that for `MultiSelection` and `SingleSelection`.

Your concern about being harder to use is also valid, but it is not much harder to use. You basically need the same information, just that a few things are now passed to the store being passed as prop, instead of being passed as props directly. On the counter side it will allow to reuse information in the store, and being generally more flexible.

Reply:
@nnatter Actually I think it makes sense for me to wait until we agree on this one, before I continue to work. Because that would probably change a lot :see_no_evil:

Reply:
you are right, i did not think about the `List`, `Form` and `SmartContent` containers. i am not sure what is the better solution here (or if there is a better solution at all). 

its absolutely okay for me to decide to always pass the stores. but in that case it would be nice to add a todo comment to the containers that work without a store prop that this should should be changed.

but how do we explain that we pass the `SelectionStore` via prop but instantiate the `SearchStore` inside of the component? whats the criteria for passing a store via prop? 🙈

Reply:
I would definitely go for always passing the stores, as I am convinced that this is the better solution, for the above reasons.

I can add some todo comments in other places, although I am afraid that this will never be fixed :see_no_evil: 

I would say the difference between the `MultiSelectionStore` and the `SearchStore` in this case is, that the `MultiSelectionStore` holds the actual data, and the `SearchStore` is just used internally and has a very high fluctation (basically a complete new set of data is a keystroke away). Also the data from the `SearchStore` is passed to the `MultiSelectionStore` anyway if necessary. That's why I would not implement that store the same way, but I think that we won't trip over this use case very often.

Reply:
alright, so if the store contains the value of a container, we will pass it to the container. if its just used for internal data, the container can instantiate it by itself.

yeah i dont think that we will ever fix that, but it might prevent to implement new component that way 🙂